pyccl/halos/massdef.py

```diff
@@ -286,3 +286,12 @@ def MassDefVir(c_m='Klypin11'):
         c_m (string): concentration-mass relation.
     """
     return MassDef('vir', 'critical', c_m_relation=c_m)
+
+
+def MassDefFof(c_m=None):
```

Comment:
It doesn't make sense to associate a for mdef with a cm relation, since these masses cannot be translated into other masses

Reply:
I know, it's only there to preserve the signature, but we will be addressing all of that in 1064, so leave it for now.

Reply:
What signature? This function didn't exist before, right?

Reply:
Signature compared to the other `MassDef`s. I don't mind removing it, as it's obvious it doesn't play a role, but note that we do the same with e.g. `ConcentrationConstant`, where the input `mass_def` is arbitrary, but we keep it as an argument to have uniform signature everywhere.

Reply:
Let's just remove this. No need to add more stuff in v2 if we're not sure we'll keep it in v3

Reply:
(meaning remove the whole mdeffof, we can introduce it in v3)

Reply:
We will 100% forget, there are too many things we'll have to remember. So I just removed `cm` as an argument for now.